Tests/PackageLoadingTests/PD_4_2_LoadingTests.swift

```diff
@@ -520,14 +519,14 @@ class PackageDescription4_2LoadingTests: PackageDescriptionLoadingTests {
           case relativePath
           case unsupportedHostname
 
-          var manifestError: ManifestParseError? {
+          var manifestError: StringError? {
```

Comment:
I always assumed `ManifestParseError` has some meaning in one of the downstream API clients, probably worth confirming this is not the case

Reply:
We can wrap it if we want, but it always seemed wrong to me for these to be manifest parsing errors since they really aren't.

Reply:
I dont have a strong feeling towards it, just making sure it does not break clients

Reply:
Hm, I am not sure this *can* break clients? They always needed to handle any potential error being thrown here since Swift doesn't have typed errors. If any client expects only a particular type of error, they have always been broken.

Reply:
I personally strongly prefer dedicated error enums instead of `StringError`. We do very little to no testing on error paths, and especially in tests it's much easier to assert on a specific error enum case than `StringError`. For that reason I prefer the previous usage of an enum case, maybe it could just be renamed or a new enum created to better reflect the nature of the error?

Reply:
Adding an enum case seems reasonable. The main reason I switched to `StringError` was that the old error type was no longer accessible and the code was already using `StringError` in other cases.

Reply:
Also note that `ManifestParseError` is not *really* a typed error in any meaningful way, it is basically just another `StringError`.

Reply:
I'm doing both now:
- added a new enum for dependency mapping errors (that also covers the existing `StringError`)
- catching the error in the parsing code to throw manifest loading errors instead